tendrl/ceph_integration/sds_sync/__init__.py

```diff
@@ -239,6 +245,213 @@ def _sync_ec_profiles(self):
+        }
+
+        if old_status != new_status:
+            if health_severity[new_status] < health_severity[old_status]:
+                # A worsening of health
+                event_sev = health_severity[new_status]
+                msg = "Health of cluster '{name}' degraded from {old} to {new}".format(
+                    old=old_status,
+                    new=new_status,
+                    name=NS.tendrl_context.cluster_name
+                )
+            else:
+                # An improvement in health
+                event_sev = INFO
+                msg = "Health of cluster '{name}' recovered from {old} to {new}".format(
+                    old=old_status,
+                    new=new_status,
+                    name=NS.tendrl_context.cluster_name
+                )
+
+            if health_severity[new_status] < INFO:
+                pass
+
+            self._emit_event(event_sev, 'health', new_status, msg)
+
+    def _on_mon_status(self, data):
+        old_quorum = NS.ceph.objects.SyncObject(
+            sync_type='mon_status'
+        ).load().data['quorum']
+        new_quorum = set(data['quorum'])
+
+        def _mon_event(severity, msg, mon_rank):
+            name = data.mons_by_rank[mon_rank]['name']
+            self._emit_event(
+                severity,
+                'monitor',
+                str(new_quorum),
+                msg.format(
+                    cluster_name=NS.tendrl_context.cluster_name,
+                    mon_name=name
+                )
+            )
+
+        for rank in new_quorum - old_quorum:
+            _mon_event(
+                INFO,
+                "Mon '{cluster_name}.{mon_name}' joined quorum",
+                rank
+            )
+
+        for rank in old_quorum - new_quorum:
+            _mon_event(
+                WARNING,
+                "Mon '{cluster_name}.{mon_name}' left quorum",
+                rank
+            )
+
+    def _on_osd_map(self, data):
+        if NS.ceph.objects.SyncObject(sync_type='osd_map').exists():
+            old_osds_det = NS.ceph.objects.SyncObject(
+                sync_type='osd_map'
+            ).load().data['osds']
+            old_osd_ids = set([o['osd'] for o in old_osds_det])
+            new_osd_ids = set([o['osd'] for o in data['osds']])
+            deleted_osds = old_osd_ids - new_osd_ids
+            created_osds = new_osd_ids - old_osd_ids
+
+            def osd_event(severity, msg, osd_id, curr_value):
+                self._emit_event(
+                    severity,
+                    'osd_status',
+                    curr_value,
+                    msg.format(
+                        name=NS.tendrl_context.cluster_name,
+                        id=osd_id
+                    )
+                )
+
+            # Generate events for removed OSDs
+            for osd_id in deleted_osds:
+                osd_event(
+                    INFO,
+                    "OSD {name}.{id} removed from the cluster map",
```

Comment:
Can you please add the cluster name to the message...
Just so that in a multi-cluster deployment it makes more value...